gsplat/rasterize.py

```diff
@@ -82,6 +83,86 @@ def rasterize_gaussians(
     )
 
 
+def rasterize_indices(
+    xys: Float[Tensor, "*batch 2"],
+    depths: Float[Tensor, "*batch 1"],
+    radii: Float[Tensor, "*batch 1"],
+    conics: Float[Tensor, "*batch 3"],
+    num_tiles_hit: Int[Tensor, "*batch 1"],
+    opacity: Float[Tensor, "*batch 1"],
+    img_height: int,
+    img_width: int,
+) -> Tensor:
+    """Rasterizes 2D gaussians by sorting and binning gaussian intersections for each tile and returns an N-dimensional output using alpha-compositing.
+
+    Note:
+        This function is not differentiable w.r.t any inputs.
+
+    Args:
+        xys (Tensor): xy coords of 2D gaussians.
+        depths (Tensor): depths of 2D gaussians.
+        radii (Tensor): radii of 2D gaussians
+        conics (Tensor): conics (inverse of covariance) of 2D gaussians in upper triangular format
+        num_tiles_hit (Tensor): number of tiles hit per gaussian
+        opacity (Tensor): opacity associated with the gaussians.
+        img_height (int): height of the rendered image.
+        img_width (int): width of the rendered image.
+
+    Returns:
+        A Tensor:
+
+        - **gaussian_ids** (Tensor): Packed (flattened) gaussian ids for intersects. [M,]
```

Comment:
can you elaborate a bit on the docstrings of return types? M is >> number of pixels right, since every gaussian gets counted once for each time it's seen in a pixel? if there's any gaurantees on the formatting of the tensors that'd be nice to note too (like it's sorted by pixel or sorted by gaussian etc)